backend/src/main/java/ai/verta/modeldb/experiment/FutureExperimentServiceImpl.java

```diff
@@ -235,7 +240,69 @@ public void getExperimentById(
   @Override
   public void getExperimentByName(
       GetExperimentByName request, StreamObserver<GetExperimentByName.Response> responseObserver) {
-    super.getExperimentByName(request, responseObserver);
+    try {
+      final var requestValidationFuture =
+          InternalFuture.runAsync(
+              () -> {
+                String errorMessage = null;
+                if (request.getProjectId().isEmpty() || request.getName().isEmpty()) {
+                  errorMessage =
+                      "Experiment name and Project ID is not found in GetExperimentByName request";
+                } else if (request.getProjectId().isEmpty()) {
+                  errorMessage = "Project ID not found in GetExperimentByName request";
+                } else if (request.getName().isEmpty()) {
+                  errorMessage = "Experiment name not found in GetExperimentByName request";
+                }
+                if (errorMessage != null) {
+                  throw new InvalidArgumentException(errorMessage);
+                }
```

Comment:
Maybe something like this instead:
```suggestion
                List<String> errorMessages = new ArrayListL<>();
                if (request.getProjectId().isEmpty()) {
                  errorMessages.add("Project ID not found in GetExperimentByName request");
                }
                if (request.getName().isEmpty()) {
                  errorMessages.add("Experiment name not found in GetExperimentByName request");
                }
                if (!errorMessages.isEmpty()) {
                  throw new InvalidArgumentException(String.join("\n", errorMessages);
                }
```

Reply:
Fixed